patchwork-capabilities/src/main/java/com/patchworkmc/impl/capability/CapabilityRegisteredCallbackInternal.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.impl.capability;
+
+import java.util.IdentityHashMap;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+import com.patchworkmc.api.capability.CapabilityRegisteredCallback;
+
+public class CapabilityRegisteredCallbackInternal {
+	private static final Logger LOGGER = LogManager.getLogger(CapabilityRegisteredCallback.class);
+	@SuppressWarnings("rawtypes")
+	private static final Map<Class<?>, Event> CALLBACKS = new IdentityHashMap<>();
```

Comment:
why not use `Event<CapabilityRegisteredCallback<C>>` as the type parameter here to avoid the warnings?

Reply:
And where would I get <C> from?

Reply:
You could just use ?, It would be a more specific type signature

Reply:
`private static final Map<Class<?>, Event<?>> CALLBACKS = new IdentityHashMap<>();`

Reply:
`CapabilityRegisteredCallback<?>`